uiuc_transportation_demand/functions.py

```diff
@@ -0,0 +1,282 @@
+
+                counter += Dict['E85'] * (float(line[16].split(' ')[0]))
+
+    allowed_fuels_function = {
+        'Gasoline': round(counter, 2),
+        'Diesel': round(counter / Dict['Diesel'], 2),
+        'E85': round(counter / Dict['E85'], 2),
+        'Hydrogen': round(counter / Dict['Hydrogen']),
+        'Electricity': round(counter / Dict['Electricity'])
+    }
+
+    amount = allowed_fuels_function[Fuel]
+
+    return amount
+
+
+def fuel_equivalent_cost(Fuel):
+
+    """
+    This function returns the total cost of operating the 2019 UIUC
+    fleet on the desired fuel technology alone.
+
+    Parameters:
+    -----------
+    Fuel: string
+        Type of fuel technology to be analyzed ('Gasoline', 'Diesel',
+        'E85', 'Hydrogen', or 'Electricity').
+
+    Returns:
+    --------
+    cost: float
+        The total cost in $ of using only that fuel technology type
+        to replace the entire 2019 UIUC fleet fuel.
+
+    """
+
+    if type(Fuel) != str:
+        raise TypeError(
+         'Please ensure that the input parameter is of type "str".'
+        )
+
+    CostDictionary = unit_cost_dictionary()
+
+    GGEDictionary = gge_dictionary()
+
+    if Fuel not in GGEDictionary:
+        raise IndexError(
+         f"{Fuel} not supported. Try 'Gasoline', 'Diesel', 'E85', 'Hydrogen', or 'Electricity'."
+        )
+
+    Fuel_Amount = fuel_equivalent(Fuel)
+
+    cost = round(Fuel_Amount * CostDictionary[Fuel], 2)
+
+    return cost
+
+
+def co2_equivalent():
+
+    """
+    This function generates a dictionary linking each fuel technology
+    type with its equivalent CO2 emissions per unit
+
+    Returns:
+    -----------
+    CO2: dict
+        A dictionary linking each of the five main technology types
+        to the equivalent amount of CO2 emitted per unit
+
+    """
+
+    CO2 = {
+        'Gasoline': 8.89,
+        'Diesel': 10.16,
+        'E85': 5.75 * 0.85 + 8.89 * 0.15,
+        'Hydrogen': 0,
+        'Electricity': 0
+    }
+
+    return CO2
+
+
+def co2_emissions(Fuel):
+
+    """
+    This function returns the total amount of equivalent CO2
+    release in kg from running the 2019 UIUC fleet off of a
+    desired fuel type alone.
+
+    Parameters:
+    -----------
+    Fuel: string
+        Type of fuel technology to be analyzed ('Gasoline', 'Diesel',
+        'E85', 'Hydrogen', or 'Electricity').
+
+    Returns:
+    --------
+    emissions: float
+        The total amount of equivalent CO2 release in kg from using only
+        the desired fuel technology to power the 2019 UIUC fleet.
+
+    """
+
+    if type(Fuel) != str:
+        raise TypeError(
+         'Please ensure that the input parameter is of type "str".'
+        )
+
+    GGEDictionary = gge_dictionary()
+
+    CO2Dictionary = co2_equivalent()
+
+    if Fuel not in GGEDictionary:
+        raise IndexError(
+         f"{Fuel} not supported. Try 'Gasoline', 'Diesel', 'E85', 'Hydrogen', or 'Electricity'."
```

Comment:
```suggestion
         f"{Fuel} not supported. Try {list(GGEDictionary.keys())}."
```
Same thing as before